galsim/real.py

```diff
@@ -259,10 +279,14 @@ def __init__(self, real_galaxy_catalog, index=None, id=None, random=False,
         if logger:
             logger.debug('RealGalaxy %d: Made original_gal',use_index)
 
-        # If flux is None, leave flux as given by original image
-        if flux is not None:
-            flux_rescale = flux / self.original_gal.getFlux()
-        if flux_rescale is not None:
+        # Only alter normalization if a change is requested
+        if flux is not None or flux_rescale is not None or normalize_area:
+            if flux_rescale is None:
+                flux_rescale = 1
+            if normalize_area:
+                flux_rescale /= HST_area
```

Comment:
Done.  Also, renamed `normalize_area` -> `area_norm` to better match new behavior.